pkg/app/server/grpcapi/grpcapi.go

```diff
@@ -166,18 +166,22 @@ func getEncriptionKey(se *model.Piped_SecretEncryption) ([]byte, error) {
 }
 
 func gRPCStoreError(err error, msg string) error {
-	switch err {
-	case nil:
+	if err == nil {
 		return nil
-	case datastore.ErrNotFound, filestore.ErrNotFound, stagelogstore.ErrNotFound:
+	}
+	if errors.Is(err, datastore.ErrNotFound) || errors.Is(err, filestore.ErrNotFound) || errors.Is(err, stagelogstore.ErrNotFound) {
 		return status.Error(codes.NotFound, fmt.Sprintf("Entity was not found to %s", msg))
-	case datastore.ErrInvalidArgument:
+	}
+	if errors.Is(err, datastore.ErrInvalidArgument) {
 		return status.Error(codes.InvalidArgument, fmt.Sprintf("Invalid argument to %s", msg))
-	case datastore.ErrAlreadyExists:
+	}
+	if errors.Is(err, datastore.ErrAlreadyExists) {
 		return status.Error(codes.AlreadyExists, fmt.Sprintf("Entity already exists to %s", msg))
-	default:
-		return status.Error(codes.Internal, fmt.Sprintf("Failed to %s", msg))
 	}
+	if errors.Is(err, datastore.ErrUserDefined) {
```

Comment:
Why do we have to change from switch case to multiple if 🤔

Reply:
If we generate an error with a message from MySQL, handling errors with a switch cannot work properly. That's why I had to change it to multi-if blocks.

Reply:
Which means the switch block can not check the type of datastore.ErrUserDefined error 🤔 It's a bit weird to me. Btw, how about adding test for this function. Looks like this function is testable without mocking anything.

Reply:
When an error message from MySQL is added to datastore.ErrUserDefined, checking with the switch block does not work because the new error is no longer the same as datastore.ErrUserDefined.

I agree with you to add tests for it 👍